### PR TITLE
Initial .NET implementation

### PR DESCRIPTION
**Pull request by YurySolovyov**
Very simple naive single-threaded implementation to use as a reference and improved upon

### CODE REVIEW

**Comment by YurySolovyov** [comment]
Speed is unimpressive:
```
File: dotnet-1684521877.csv
Processed- 1000000
sec- 0; TH- 4461
sec- 1; TH- 17489
sec- 2; TH- 19365
sec- 3; TH- 19189
sec- 4; TH- 19088
sec- 5; TH- 19122
sec- 6; TH- 19100
sec- 7; TH- 19189
sec- 8; TH- 19153
sec- 9; TH- 17820
sec- 10; TH- 18763
sec- 11; TH- 19331
sec- 12; TH- 19166
sec- 13; TH- 18957
sec- 14; TH- 19420
sec- 15; TH- 18993
sec- 16; TH- 18117
sec- 17; TH- 17860
sec- 18; TH- 18270
sec- 19; TH- 18705
sec- 20; TH- 18785
sec- 21; TH- 19261
sec- 22; TH- 18692
sec- 23; TH- 18860
sec- 24; TH- 19659
sec- 25; TH- 18762
sec- 26; TH- 17947
sec- 27; TH- 19299
sec- 28; TH- 18299
sec- 29; TH- 18525
sec- 30; TH- 19093
sec- 31; TH- 19879
sec- 32; TH- 19767
sec- 33; TH- 19117
sec- 34; TH- 19195
sec- 35; TH- 19346
sec- 36; TH- 18739
sec- 37; TH- 19123
sec- 38; TH- 19131
sec- 39; TH- 18623
sec- 40; TH- 19533
sec- 41; TH- 19328
sec- 42; TH- 19545
sec- 43; TH- 18029
sec- 44; TH- 18802
sec- 45; TH- 19053
sec- 46; TH- 19361
sec- 47; TH- 18763
sec- 48; TH- 18789
sec- 49; TH- 18539
sec- 50; TH- 19948
sec- 51; TH- 19864
sec- 52; TH- 19704
sec- 53; TH- 9082

Avg items/s- 18970
Min items/s- 17489
Max items/s- 19948
Signature- 24b760abba866d98a760b0cac44543b2
Expected - 24b760abba866d98a760b0cac44543b2
```

**Comment by YurySolovyov** [comment]
Simple threading gives huge boost:
```
File: dotnet-1684693080.csv
Processed- 1000000
sec- 0; TH- 111562
sec- 1; TH- 168605
sec- 2; TH- 168496
sec- 3; TH- 168252
sec- 4; TH- 163888
sec- 5; TH- 196597
sec- 6; TH- 22600

Avg items/s- 173167
Min items/s- 163888
Max items/s- 196597
Signature- 24b760abba866d98a760b0cac44543b2
Expected - 24b760abba866d98a760b0cac44543b2
```

File: dotnet/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM mcr.microsoft.com/dotnet/sdk:7.0-alpine3.17
```
Comment (kirill-komarovich) [inline dotnet/Dockerfile:1]:
```suggestion
FROM mcr.microsoft.com/dotnet/sdk:7.0-alpine3.16
```

File: dotnet/Program.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+using System.Globalization;
+
+using CsvHelper;
+using CsvHelper.Configuration;
+using StackExchange.Redis;
+
+var host = Environment.GetEnvironmentVariable("REDIS_HOST") ?? "redis";
+var redis = ConnectionMultiplexer.Connect(host).GetDatabase();
+
+var threads = Int32.Parse(Environment.GetEnvironmentVariable("WORKERS") ?? "16");
```
Comment (kirill-komarovich) [inline dotnet/Program.cs:17]: can it get available cores and use it?

Comment (YurySolovyov) [reply dotnet/Program.cs]: Done

**Pull request merged**